### PR TITLE
[SPARK-7485] [build] Remove pyspark files from assembly.

### PR DESCRIPTION
**Pull request by vanzin**
The sbt part of the build is hacky; it basically tricks sbt
into generating the zip by using a generator, but returns
an empty list for the generated files so that nothing is
actually added to the assembly.

### CODE REVIEW

**Comment by srowen** [comment]
LGTM, certainly the Maven parts. I take your word on the SBT part.

File: project/SparkBuild.scala [hunk 1]
```diff
@@ -388,7 +387,8 @@ object PySparkAssembly {
       if (!dst.isDirectory()) {
```
Comment (lianhuiwang) [inline project/SparkBuild.scala]: i think these codes also can be removed.

**Comment by tgravescs** [comment]
looks good other then the one comment about removing the dst directory stuff from sbt

**Comment by andrewor14** [comment]
LGTM. Great to see so much code removed!

**Comment by mengxr** [comment]
@vanzin Does it mean that Maven users have to use sbt to include Python support?

**Comment by vanzin** [comment]
No. See the bug referenced in this PR.

**Comment by andrewor14** [comment]
Merging into master 1.4.

**Pull request closed without merging**